### PR TITLE
Added `Add` button in `Select`

### LINKED ISSUE #1968: Update Select component to include `Add` button
Ref: https://github.com/bigbinary/neeto-engineering-ui-ux/issues/202

@ajmaln _a

### PR DESCRIPTION
**Pull request by ajmaln**
- Fixes #1968 

**Description**
Added: `Add` button in _Select_

**Checklist**

- [x] I have made corresponding changes to the documentation.
- [x] I have updated the types definition of modified exports.
- [x] I have verified the functionality in some of the neeto web-apps.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary label (patch/minor/major - If package publish
      is required).

**Reviewers**

### CODE REVIEW

**Comment by ajmaln** [comment]
@praveen-murali-ind _a Please have a look

**Comment by praveen-murali-ind** [comment]
@ajmaln I have made the modifications discussed. Please check.

fbe903ac-8192-4d06-86bd-2a34ecbaafe7

Disabled state

Screenshot 2023-11-08 at 9 33 14 PM

File: index.d.ts [hunk 1]
```diff
@@ -332,6 +332,8 @@ export type SelectProps = {
   fetchMore?: () => void;
   totalOptionsCount?: number;
   isAsyncLoadOptionEnabled?: boolean;
+  isMulti?: boolean;
+  addButtonLabel: string;
```
Comment (ajmaln) [inline index.d.ts:336]:
```suggestion
  addButtonLabel?: string;
```

**Comment by ajmaln** [comment]
@praveen-murali-ind Looks good

**Pull request merged**